pipeline/run_stability_tests.sh

```diff
@@ -63,31 +57,50 @@ EOF
 echo -e ":wait\n:exit\n" \
  >> stability_test/stability_tests.txt
 
-#./run.sh stability_test/stability_tests.txt
 [ -f libblastjni.so ] || gsutil cp gs://blast-lib/libblastjni.so .
 spark-submit --master yarn \
     "$LOG_CONF" \
     --class $BC_CLASS $BC_JAR $BC_INI stability_test/stability_tests.txt
 
 
 cd report || exit
-grep -h "done at" ./*.txt | sort > dones &
+grep -h "done at" ./*.txt | sort > dones
 for asn in *.asn1; do
     asntool -m ../../lib_builder/asn.all \
         -t Seq-annot -d "$asn" -p "$asn.txt"
 done
 
-wait
-
-#rm -f ../*.result
-#wc -l -- *.asn1 | sort > ../asn1.wc.result
 unset LC_ALL # Messes with sorting
 wc -l ./*.asn1.txt | awk '{print $1 "\t" $2;}' | sort -k2 > ../asn1.txt.wc.result
-#if diff asn1.wc.expected asn1.wc.result; then
-#    echo "Differences in .asn1 output"
-#fi
+
+DATE=$(date "+%Y%m%d%H%M")
+echo "Uploading test results to gs://blast-stability-test-results/$DATE"
+gsutil -m cp -r ./* "gs://blast-stability-test-results/$DATE" > /dev/null 2>&1
+echo "Uploaded  test results"
 
 cd ..
 if diff asn1.txt.wc.expected asn1.txt.wc.result; then
     echo "Differences in .asn1.txt output"
 fi
+
+echo "Downloading reference results..."
+gsutil -m cp -nr gs://blast-results-reference/ . > /dev/null 2>&1
+numrefs=$(find blast-results-reference/ -name "*.gz" | wc -l)
+echo "Downloaded $numrefs reference results."
+
+echo "Uncompressing reference results..."
+cd blast-results-reference || exit
+find ./ -name "*gz" -print0 | nice xargs -0 -n 8 -P 8 gunzip
+echo "Uncompressed  reference results."
+cd ..
+
+for check in blast-results-reference/*.asn; do
+    base=$(basename "$check")
+    req="report/REQ_${base}1.txt"
+    diff "$check" "$req" > /dev/null 2>&1
```

Comment:
I don't think this will work as the reference results are binary ASN.1

Reply:
Agree, diff doesn't appear useful at the moment, need a proper blast diffing tool eventually.
The uncompressed reference results are Blast4-archives, not Seq-annots? https://jira.ncbi.nlm.nih.gov/browse/WB-3344

```
Blast4-archive ::= {
  request {
    ident "2.8.1+",
    body queue-search {
      program "blastn",
```

Reply:
Yes.